include/tables/scene_table.h

```diff
@@ -0,0 +1,229 @@
+/**
+ * Scene Table
+ *
+ * DEFINE_SCENE should be used for most scenes
+ *    - Argument 0: Name of the scene segment in spec
+ *    - Argument 1: Enum value for this scene
+ *    - Argument 2: Title card's text id, or zero for no title card
+ *    - Argument 3: Scene draw config index
+ *    - Argument 4: Restriction flags. See `RESTRICTIONS_SET` on z_parameter.c
```

Comment:
I think because there would be issues if a scene is missing its restriction flags, I'm fine with having it in the scene table